x/sequencer/keeper/rotation.go

```diff
@@ -95,3 +105,44 @@ func (k Keeper) OnProposerLastBlock(ctx sdk.Context, proposer types.Sequencer) e
 	)
 	return nil
 }
+
+// setSuccessorForRollapp will assign a successor. It won't replace an existing one.
+// It will prioritize non sentinel
+func (k Keeper) setSuccessorForRollapp(ctx sdk.Context, rollapp string) error {
```

Comment:
this seems like a repeat of part of the `RecoverFromHaltIfNeeded` function.
also I suggest for standartization `setSuccessorForRollapp` -> `chooseSuccessor`